python/ray/tune/execution/experiment_state.py

```diff
@@ -138,6 +137,13 @@ def __init__(
         # Upload triggered by trial checkpoints
         self._sync_every_n_trial_checkpoints = sync_every_n_trial_checkpoints
         self._trial_num_checkpoints_since_last_sync: Dict[Trial, int] = Counter()
+
+        self._slow_sync_threshold = float(
+            os.environ.get(
+                "TUNE_WARN_SLOW_EXPERIMENT_CHECKPOINT_SYNC_THRESHOLD_S", "30"
```

Comment:
I am wondering if we should always warn users if they don't have `fsspec`, rather than just if their exp syncing is taking a while? The threshold might mask the problems and the user would never know to do something.

Will there be a separate warning checking if using any cloud storage + no `fsspec`?

Reply:
Yes! I have added the warnings in the follow-up PR: https://github.com/ray-project/ray/pull/34663